float64 enc1
float64 enc2
float64 enc3